code/_onclick/hud/alert.dm

```diff
@@ -59,7 +59,7 @@
 		master_appearance.pixel_x = new_master.base_pixel_x
 		master_appearance.pixel_y = new_master.base_pixel_y
 		master_appearance.pixel_z = new_master.base_pixel_z
-		thealert.add_overlay(master_appearance)
+		thealert.add_overlay(strip_light_underlays(master_appearance))
```

Comment:
instead of stripping just lighting you should strip anything that is not on the same plane as the initial appearance (will not be carried into the float)

Reply:
(please don't mark things as resolved if you ask me a question reee)
yeah, also it needs to fully denest overlays, which is.... very annoying. because insertion order ocntrols overlays you basically need to unwrap every component and check, then chekc its children, then reapply bottom up.

of note, this may require something similar to the below to prevent hell bullshit problems (miscaching or some shit? byond bug I never made a testcase for)
![image](https://github.com/user-attachments/assets/4b0c84a5-bb3b-454f-a61f-68f4150b4a36)

Reply:
oh... agh that seems like a pain.